src/Illuminate/Bus/Queueable.php

```diff
@@ -95,7 +140,19 @@ public function dispatchNextJobInChain()
     {
         if (! empty($this->chained)) {
             dispatch(tap(unserialize(array_shift($this->chained)), function ($next) {
+                if (in_array('Illuminate\Bus\Queueable', class_uses_recursive($next))) {
+                    /* @var \Illuminate\Bus\Queueable $next */
+                } else {
+                    throw new \Exception('Trying to dispatch an object that is not Queueable');
+                }
+                // pass the chain settings on to the next job in the chain, IF this job does not have a new chain settings set...
+                $next->onChainConnection($next->chain_connection ?: $this->chain_connection);
+                $next->onChainQueue($next->chain_queue ?: $this->chain_queue);
+                // array of remaining jobs...
                 $next->chained = $this->chained;
+                // use the chain setting if this job is not specifically set.
+                $next->onConnection($next->connection ?: $this->chain_connection);
```

Comment:
You said this doesn't change existing behavior but this appears to do so. Why are you using the chain connection here?

Reply:
The default chain_connection is null. So is the next job connection.
If you set the chain connection, then the next job should use it as its connection if is it not specifically set on that job. I wrote test cases for these things also.